service/matching/physical_task_queue_manager.go

```diff
@@ -364,8 +368,8 @@ func (c *physicalTaskQueueManagerImpl) PollTask(
 		task.namespace = c.partitionMgr.ns.Name()
 		task.backlogCountHint = c.backlogMgr.BacklogCountHint
 
-		if task.redirectInfo == nil && (!task.isStarted() || !task.started.hasEmptyResponse()) {
-			// only track the original polls, not forwarded ones.
+		if pollMetadata.forwardedFrom == "" && // only track the original polls, not forwarded ones.
+			(!task.isStarted() || !task.started.hasEmptyResponse()) { // Need to filter out the empty "started" ones
```

Comment:
why do we not want to track the empty "started" ones as dispatched?

Reply:
because that means no task was dispatched.